### PR TITLE
CAMEL-20185 - Kubernetes CronJob Component 

### PR DESCRIPTION
**Pull request by oscerd**
# Description

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

File: components/camel-kubernetes/src/main/docs/kubernetes-cronjob-component.adoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+:component-header: Both producer and consumer are supported
+//Manually maintained attributes
+:group: Kubernetes
+:camel-spring-boot-name: kubernetes
+
+*Since Camel {since}*
+
+*{component-header}*
+
+The Kubernetes CronJob component is one of xref:kubernetes-summary.adoc[Kubernetes Components] which
```
[the remark this answers is not in the document]
Comment (oscerd) [reply components/camel-kubernetes/src/main/docs/kubernetes-cronjob-component.adoc:18]: I guess it's a copy paste and it's wrong even in other places. I'll check.

File: components/camel-kubernetes/src/main/java/org/apache/camel/component/kubernetes/cronjob/KubernetesCronJobProducer.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                        .resource(cronjobCreating));
+
+        prepareOutboundMessage(exchange, cronJob);
+    }
+
+    protected void doDeleteCronJob(Exchange exchange) {
+        String cronjobName = exchange.getIn().getHeader(KubernetesConstants.KUBERNETES_CRON_JOB_NAME, String.class);
+        String namespaceName = exchange.getIn().getHeader(KubernetesConstants.KUBERNETES_NAMESPACE_NAME, String.class);
+        if (ObjectHelper.isEmpty(cronjobName)) {
+            LOG.error("Delete a specific cronjob require specify a cronjob name");
```
[the remark this answers is not in the document]
Comment (oscerd) [reply components/camel-kubernetes/src/main/java/org/apache/camel/component/kubernetes/cronjob/KubernetesCronJobProducer.java:171]: Same, it needs to be reviewed everywhere.

File: components/camel-kubernetes/src/main/java/org/apache/camel/component/kubernetes/cronjob/KubernetesCronJobEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import org.apache.camel.component.kubernetes.KubernetesConfiguration;
+import org.apache.camel.component.kubernetes.KubernetesConstants;
+import org.apache.camel.spi.UriEndpoint;
+
+import static org.apache.camel.component.kubernetes.KubernetesConstants.SCHEME_CRON_JOB;
+
+/**
+ * Perform operations on Kubernetes CronJob.
+ */
+@UriEndpoint(firstVersion = "4.3.0", scheme = SCHEME_CRON_JOB, title = "Kubernetes Cronjob",
```
[the remark this answers is not in the document]
Comment (oscerd) [reply components/camel-kubernetes/src/main/java/org/apache/camel/component/kubernetes/cronjob/KubernetesCronJobEndpoint.java:33]: so it's false.

[the remark this answers is not in the document]
Comment (davsclaus) [reply components/camel-kubernetes/src/main/java/org/apache/camel/component/kubernetes/cronjob/KubernetesCronJobEndpoint.java:33]: Ah yeah my bad, should be producerOnly = true

**Comment by oscerd** [comment]
It should be ok now.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**